File: Microsoft.Toolkit.Uwp.Input.GazeInteraction/GazeEventArgs.cpp
//Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license.
//See LICENSE in the project root for license information.

#include "pch.h"
#include "GazeEventArgs.h"
#include "GazeEventArgs.g.cpp"

namespace winrt::Microsoft::Toolkit::Uwp::Input::GazeInteraction::implementation
{
	GazeEventArgs::GazeEventArgs() :
		m_handled{ false },
		m_location{},
		m_timestamp{}
	{
	}

	bool GazeEventArgs::Handled()
	{
		return m_handled;
	}

	void GazeEventArgs::Handled(bool value)
	{
		m_handled = value;
	}

	Windows::Foundation::Point GazeEventArgs::Location()
	{
		return m_location;
	}

	Windows::Foundation::TimeSpan GazeEventArgs::Timestamp()
	{
		return m_timestamp;
	}

	void GazeEventArgs::Set(Windows::Foundation::Point const& location, Windows::Foundation::TimeSpan const& timestamp)
	{
		m_handled = false;
		m_location = location;
		m_timestamp = timestamp;
	}
}

File: Microsoft.Toolkit.Uwp.Input.GazeInteraction/GazeEventArgs.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using System;
using Windows.Foundation;

namespace Microsoft.Toolkit.Uwp.Input.GazeInteraction
{
    /// <summary>
    /// EventArgs used to send Gaze events. See <see cref="GazePointer.GazeEvent"/>
    /// </summary>
    public class GazeEventArgs
    {
        /// <summary>
        /// Gets or sets a value indicating whether the application handled the event. If this parameter is set to true, the library prevents handling the events on that particular gaze event.
        /// </summary>
        public bool Handled { get; set; }

        /// <summary>
        /// Gets or sets the location of the Gaze event
        /// </summary>
        public Point Location { get; set; }

        /// <summary>
        /// Gets or sets the timestamp of the gaze event
        /// </summary>
        public TimeSpan Timestamp { get; set; }

        internal void Set(Point location, TimeSpan timestamp)
        {
            Handled = false;
            Location = location;
            Timestamp = timestamp;
        }
    }
}
